src/components/suggestion/SuggestionSearch.tsx

```diff
@@ -0,0 +1,103 @@
+import { ListSuggestSearchFromVinted } from '../../api/api';
+import React, { useEffect, useState } from 'react';
+import CardSuggestionSearch from './CardSuggestionSearch';
+import { ChevronRight, ChevronLeft } from 'lucide-react';
+
+const SuggestionSearch: React.FC = () => {
+  const [scrollCards, setScrollCard] = useState<'left' | 'right' | 'both'>('right');
+  const [scrollValue, setScrollValue] = useState<number>(0);
+  const divToScroll = document.getElementById('showScroll');
+  const divSup = document.getElementById('divSup');
+
+  
+  const [products, setProducts] = useState<
+  typeof ListSuggestSearchFromVinted | []
+  >([]);
+  
+  useEffect(() => {
+    setProducts(ListSuggestSearchFromVinted);
+  }, []);
+  
+  
+  divToScroll?.addEventListener('scroll', () => {
+    setScrollValue(divToScroll?.scrollLeft);
+  });
+
+
+
+  useEffect(() => {
+    // Calcul de la taille max du scroll en fonction de la taille de l'ecran
+    const clientWidth = divSup?.clientWidth;
+    let tailleScroll = 1280
+    if (clientWidth) {
+      tailleScroll = 1280 - clientWidth;
+    }
+
+    if(scrollValue === 0) {
+      setScrollCard('left');
+    }
+    else if (scrollValue === tailleScroll) {
+      setScrollCard('right');
+    } 
+    else {
+      setScrollCard('both');
+    }
+  }, [scrollValue]);
+
+  
+  // LOGIQUE AU CLIC SU BTN
+  const handleClicRight = () => {
+    divToScroll?.scrollTo({
+      left: 1400,
+      behavior: 'smooth',
+    });
+  };
+  const handleClicLeft = () => {
+    divToScroll?.scrollTo({
+      left: 0,
+      behavior: 'smooth',
+    });
+  };
+  
+  console.log('render')
+  return (
+    //TODO: Supprimer container a l'integration
+    <div className="container mx-auto py-12 max-w-[1240px]">
+      <h2 className="text-2xl pb-4">Suggestions de recherche</h2>
+      <div className="relative h-[75px] overflow-hidden" id="divSup">
+        <div
+          className="flex h-[90px] overflow-x-auto overflow-y-hidden"
+          id="showScroll"
+        >
+          <div className="flex shrink-0 flex-nowrap">
+            {products.map((product) => (
+              <CardSuggestionSearch key={product.marque} product={product} />
+            ))}
+          </div>
+
+          {scrollCards === 'right' ||
+            ('both' && (
+              <div
+                className="absolute top-5 right-2 flex items-center justify-center h-8 w-8 rounded-full bg-gray-600 opacity-40 cursor-pointer"
+                onClick={handleClicRight}
```

Comment:
Messages de Sonarlint : 
- Avoid non-native interactive elements. If using native HTML is not possible, add an appropriate role and support for tabbing, mouse, keyboard, and touch inputs to an interactive content element.sonarlint(typescript:S6848)
- Visible, non-interactive elements with click handlers must have at least one keyboard listener.

Je pense que tu dois mettre le onClick sur le composant Chevron comme ça : 

```tsx
{scrollCard === 'right' ||
('both' && (
    <div className="absolute top-5 right-2 flex items-center justify-center h-8 w-8 rounded-full bg-gray-600 opacity-40 cursor-pointer">
    <ChevronRight
        className="text-vintedBackground"
        onClick={handleClicRight}
    />
    </div>
))}
```

Reply:
Corrigé

Reply:
onClick est encore sur la div contenant le Chevron et non sur le Chevron

Reply:
perso je prefere le laisser sur la div, ce qui permet d'avoir une zone cliquable plus large surtout en mobile quand tu cliques dessus si plus tard on ajoute du padding

Reply:
et j'ai pas ce message de sonarlint pourtant il est bien activé

Reply:
Pour le 1er message de Sonarlint, (Avoid non-native interactive elements. If using native HTML is not possible, add an appropriate role and support for tabbing, mouse, keyboard, and touch inputs to an interactive content element.sonarlint(typescript:S6848), s'il faut laisser un élément autour du chevron, dans ce cas, autant que ça soit un bouton 
Pour le 2nd message de Sonarlint, il dit qu'un élément non-intéractif comme une div doit avec un listener sur le clavier qui fait la même chose que le `onClick` pour gérer la navigation via le clavier

Cela donnerait donc selon moi : 

```tsx
      <button className={`${styleDiv} right-2`} onClick={handleClicRight}>
        <ChevronRight className="text-vintedBackground" />
      </button>
```